client/my-sites/checkout/checkout/composite-checkout.js

```diff
@@ -833,6 +833,8 @@ function getCheckoutEventHandler( reduxDispatch ) {
 						currency: action.payload.total.amount.currency,
 						is_signup: action.payload.responseCart.is_signup,
 						products: action.payload.responseCart.products,
+						coupon_code: action.payload.couponItem?.wpcom_meta.couponCode ?? '',
```

Comment:
Just to note that this will conflict with #40593 which moves this file into a new directory. So one of us (it's a race! 😂 ) will have to rebase to make sure this gets maintained.